### PR TITLE
Trigger performance test on pr

### PR DESCRIPTION
**Pull request by Tulsishah**
- Added build.sh script which under presubmit_test folder which will  do following task.
  1. install go, fio and git.
  2. mount bucket for master branch
  3. run fio test 
  4. checkout PR branch code
  5. mount bucket for PR branch
  6. run fio test
- presubmit.cfg will give path to build.sh script
- Added job config for 256kb, 50Mb, 1Gb filesizes (Sequential Read, Sequential Write, Random Reads, Random Writes)
- Python script to fetch results from json file.

Results will print like this.

Filesize: 0.25MiB
read bw: 215.93MiB/s
write bw: 229.32MiB/s
randread bw: 15.88MiB/s
randwrite bw: 240.51MiB/s

Filesize: 48.828MiB
read bw: 786.1MiB/s
write bw: 189.2MiB/s
randread bw: 353.33MiB/s
randwrite bw: 211.29MiB/s

Filesize: 976.562MiB
read bw: 758.84MiB/s
write bw: 189.33MiB/s
randread bw: 322.52MiB/s
randwrite bw: 187.96MiB/s

### CODE REVIEW

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: perfmetrics/scripts/presubmit_test/pr_test/build.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit_test/pr_test/build.sh:1]: I thought in future if we want to add some other presubmit tests, as we have multiple [continuous_test](https://github.com/GoogleCloudPlatform/gcsfuse/tree/master/perfmetrics/scripts/continuous_test)

File: perfmetrics/scripts/show_results.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  argv = sys.argv
+  if len(argv) != 2:
+    raise TypeError('Incorrect number of arguments.\n'
+                    'Usage: '
+                    'python3 show_results.py <fio output json filepath>')
+
+  # Fetching metrics from json file
+  fio_metrics_obj = FioMetrics()
+  data = fio_metrics_obj.get_metrics(argv[1])
+  for d in data :
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/show_results.py:14]: added

File: perfmetrics/scripts/job_files/seq_rand_read_write_presubmit.fio [hunk 3]
```diff
@@ -0,0 +1,118 @@
+[global]
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/job_files/seq_rand_read_write_presubmit.fio]: fixed

File: perfmetrics/scripts/show_results.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from fio.fio_metrics import FioMetrics
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/show_results.py]: fixed

File: perfmetrics/scripts/presubmit_test/pr_test/build.sh [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+pip install google-cloud
+pip install google-cloud-vision
+pip install google-api-python-client
+echo Installing go-lang 1.19.5
+wget -O go_tar.tar.gz https://go.dev/dl/go1.19.5.linux-amd64.tar.gz
+sudo rm -rf /usr/local/go && tar -xzf go_tar.tar.gz && sudo mv go /usr/local
+export PATH=$PATH:/usr/local/go/bin
+echo Installing fio
+sudo apt-get install fio -y
+cd "${KOKORO_ARTIFACTS_DIR}/github/gcsfuse/perfmetrics/scripts"
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit_test/pr_test/build.sh]: fixed

File: perfmetrics/scripts/presubmit_test/pr_test/build.sh [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+BUCKET_NAME=gcs-fuse-loadtest
+MOUNT_POINT=gcs
+# The VM will itself exit if the gcsfuse mount fails.
+go run ../../. $GCSFUSE_FLAGS $BUCKET_NAME $MOUNT_POINT
+# Running FIO test
+chmod +x run_load_test_on_presubmit.sh
+./run_load_test_on_presubmit.sh
+sudo umount gcs
+cd ../../
+echo '[remote "origin"]
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit_test/pr_test/build.sh:39]: fixed

File: perfmetrics/scripts/run_load_test_on_presubmit.sh [hunk 7]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/run_load_test_on_presubmit.sh]: fixed

File: perfmetrics/scripts/job_files/seq_rand_read_write_presubmit.fio [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+[11_thread]
+stonewall
+startdelay=710
+directory=gcs/1gb
+filesize=1G
+rw=randread
+numjobs=40
+
+[12_thread]
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/job_files/seq_rand_read_write_presubmit.fio]: fixed

File: perfmetrics/scripts/presubmit_test/pr_test/build.sh [hunk 9]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# It will take approx 1.5-2 hours to run the script.
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit_test/pr_test/build.sh]: Updated

File: perfmetrics/scripts/presubmit/print_results.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+# For fetching fio_metrics from fio
+sys.path.append("./perfmetrics/scripts/")
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/presubmit/print_results.py:17]: how is fio_metrics getting gsheet reference without this line
https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/perfmetrics/scripts/fio/fio_metrics.py#L19

File: perfmetrics/scripts/presubmit_test/git_pr_test/build.sh [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+sudo rm -rf /usr/local/go && tar -xzf go_tar.tar.gz && sudo mv go /usr/local
+export PATH=$PATH:/usr/local/go/bin
+echo Installing fio
+sudo apt-get install fio -y
+
+cd "${KOKORO_ARTIFACTS_DIR}/github/gcsfuse"
+echo Mounting gcs bucket for master branch
+mkdir -p gcs
+GCSFUSE_FLAGS="--implicit-dirs --max-conns-per-host 100 --experimental-enable-storage-client-library=true --disable-http2"
+BUCKET_NAME=gcs-fuse-loadtest
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/presubmit_test/git_pr_test/build.sh:25]: can we change the bucket name to presubmit-perf-test

File: perfmetrics/scripts/presubmit_test/git_pr_test/presubmit.cfg [hunk 12]
```diff
@@ -0,0 +1 @@
+build_file: "gcsfuse/perfmetrics/scripts/presubmit_test/git_pr_test/build.sh"
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/presubmit_test/git_pr_test/presubmit.cfg:1]: should we call it pr_perf_test instead of git_pr_test

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: perfmetrics/scripts/presubmit/print_results.py (hunk 10 above)
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit/print_results.py:17]: we are not using gsheet in our python script.
We are using only [get_metrics](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/perfmetrics/scripts/fio/fio_metrics.py#L437) function which is returning list from json file.

File: perfmetrics/scripts/presubmit/print_results.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+                    'python3 show_results.py <fio output json filepath>')
+
+  # Fetching metrics from json file
+  fio_metrics_obj = FioMetrics()
+  data = fio_metrics_obj.get_metrics(argv[1])
+  # Iterating through data
+  for d in data :
+    # Print filesize only once
+    if d['params']['rw'] == "read":
+      print("Filesize: "+ str(round(d["params"]["filesize_kb"]/1024.0,3)) + "MiB")
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit/print_results.py]: fixed

File: perfmetrics/scripts/presubmit_test/git_pr_test/build.sh (hunk 11 above)
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit_test/git_pr_test/build.sh]: fixed

File: perfmetrics/scripts/presubmit_test/git_pr_test/presubmit.cfg (hunk 12 above)
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit_test/git_pr_test/presubmit.cfg]: fixed

File: perfmetrics/scripts/presubmit/print_results.py [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+                    'python3 show_results.py <fio output json filepath>')
+
+  # Fetching metrics from json file
+  fio_metrics_obj = FioMetrics()
+  data = fio_metrics_obj.get_metrics(argv[1])
+  # Iterating through data
+  for d in data :
+    # Print filesize only once
+    if d['params']['rw'] == "read":
+      print("\n"+"Filesize: "+ str(round(d["params"]["filesize_kb"]/1024.0,3)) + "MiB")
```
Comment (sethiay) [inline perfmetrics/scripts/presubmit/print_results.py:34]: nit: use str.format() for better readability.

**Review by sethiay — APPROVED** [review]

File: perfmetrics/scripts/presubmit/print_results.py (hunk 14 above)
Comment (Tulsishah) [reply perfmetrics/scripts/presubmit/print_results.py]: fixed

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**